manager/controllers/app/m4dapplication_controller.go

```diff
@@ -108,7 +108,9 @@ func (r *M4DApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	if hasError(applicationContext) {
 		log.Info("Reconciled with errors: " + getErrorMessages(applicationContext))
 	}
-	if !applicationContext.Status.Ready {
+
+	// trigger a new reconcile if required (the m4dapplication is not ready and not empty)
+	if !applicationContext.Status.Ready && len(applicationContext.Spec.Data) > 0 {
```

Comment:
You could write a filter so that the reconcile loop would not be called at all in this situation. For an example, see blueprint_manager.go SetupWithManager (line 331)

Reply:
We do need a reconcile loop in this situation, to delete older resources. But only once, i.e. no need to return requeue = true in this case.